src/bindings/client.ts

```diff
@@ -0,0 +1,255 @@
+import { xdr } from "@stellar/stellar-base";
+import { Spec } from "../contract";
+
+/**
+ * Generates TypeScript client class for contract methods
+ */
+export class ClientGenerator {
+  private spec: Spec;
+  private interfaceImports: Set<string> = new Set<string>();
+
+  constructor(spec: Spec) {
+    this.spec = spec;
+  }
+
+  /**
+   * Generate client class
+   */
+  generate(): string {
+    // Generate constructor method if it exists
+    let deployMethod = "";
+    try {
+      const constructorFunc = this.spec.getFunc("__constructor");
+      deployMethod = this.generateDeployMethod(constructorFunc);
+    } catch {
+      // For specs without a constructor, generate a deploy method without params
+      deployMethod = this.generateDeployMethod(undefined);
+    }
+    // Generate interface methods
+    const interfaceMethods = this.spec
+      .funcs()
+      .filter((func) => func.name().toString() !== "__constructor")
+      .map((func) => this.generateInterfaceMethod(func))
+      .join("\n");
+
+    // Build imports
+    const typeImports =
+      Array.from(this.interfaceImports).length > 0
+        ? `import { ${Array.from(this.interfaceImports).join(", ")} } from './types.js';`
+        : "";
+
+    const specEntries = this.spec.entries.map(
+      (entry) => `"${entry.toXDR("base64")}"`,
+    );
+
+    const fromJSON = this.spec
+      .funcs()
+      .filter((func) => func.name().toString() !== "__constructor")
+      .map((func) => this.generateFromJSONMethod(func))
+      .join(",");
+
+    return `import {
+  AssembledTransaction,
+  Client as ContractClient,
+  ClientOptions as ContractClientOptions,
+  Result,
+  MethodOptions,
+  Spec,
+} from '@stellar/stellar-sdk/contract';
+${typeImports}
+import { Buffer } from 'buffer';
+export interface Client {
+${interfaceMethods}
+}
+
+export class Client extends ContractClient {
+  constructor(public readonly options: ContractClientOptions) {
+    super(
+      new Spec([${specEntries.join(", ")}]),
+      options
+    );
+  }
+
+ ${deployMethod}
+  public readonly fromJSON = {
+  ${fromJSON}
+  };
+}`;
+  }
+
+  /**
+   * Generate interface method signature
+   */
+  private generateInterfaceMethod(func: any): string {
+    const name = func.name().toString();
+    const inputs = func.inputs().map((input: any) => ({
+      name: input.name().toString(),
+      type: this.generateTypeFromTypeDef(input.type()),
+    }));
+    const outputType =
+      func.outputs().length > 0
+        ? this.generateTypeFromTypeDef(func.outputs()[0])
+        : "void";
+
+    const params = this.formatMethodParameters(inputs);
+
+    return `  ${name}(${params}): Promise<AssembledTransaction<${outputType}>>;`;
```

Comment:
Should we also inject the docs pulled from the spec for each method?

Reply:
💯